R/step-colorder.R

```diff
@@ -1,11 +1,23 @@
-step_colorder <- function(parent, col_order) {
-  stopifnot(is_step(parent))
-  stopifnot(is.character(col_order))
+step_colorder <- function(x, col_order) {
+  stopifnot(is_step(x))
+  stopifnot(is.character(col_order) || is.integer(col_order))
 
-  step_call(parent,
+  if (is.integer(col_order)) {
+    if (all(col_order == seq_along(col_order))) {
```

Comment:
I think that should use `identical()` since `==` is vectorised?

Reply:
And maybe move the check until after handling the different types? i.e. `if (identical(vars, x$vars)) ...`